src/president/deck.js

```diff
@@ -0,0 +1,92 @@
+exports.deck_json =
+`{
+  "colors" : ["c","d","h","s"],
+  "cards" : ["1","2","3","4","5","6","7","8","9","10","11","12","13"]
+}`
+
+
+/*
+ * Create a deck, from the deck described in the object.
+ */
+exports.initDeck = function (object) {
+  var deck_obj = JSON.parse(object);
+  var colors = deck_obj.colors;
+  var cards = deck_obj.cards;
+
+  var deck = [];
+  colors.forEach(color => {
+    cards.forEach(card => {
+      deck.push(card + color);
+    })
+  });
+  return deck;
+}
+
+/*
+ * Return the empty deck.
+ */
+exports.emptyDeck = function (deck) {
+  return [];
+}
+
+/*
+ * Return true if the deck is empty.
+ */
+exports.isEmptyDeck = function (deck) {
+  return deck.length == 0;
+}
+
+/*
+ * Remove card from deck,
+ * if the card is not in the deck, nothing happens.
+ */
+exports.removeCardFromDeck = function (deck, card) {
+  const index = deck.indexOf(card);
+  if (index > -1) {
+    deck.splice(index, 1);
+  }
+}
+
+/*
+ * Remove cards from the deck,
+ * if the cards are not in the deck, nothing happens.
+ */
+exports.removeCardsFromDeck = function (deck, cards) {
+  cards.forEach(card => exports.removeCardFromDeck(deck, card));
+}
+
+/*
+ * Add card to the deck.
+ */
+exports.addCardToDeck = function (deck, card) {
+  deck.push(card);
+}
+
+/*
+ * Add cards to the deck.
+ */
+exports.addCardsToDeck = function (deck, cards) {
+  cards.forEach(card => exports.addCardToDeck(deck, card));
+}
+
+/*
+ * Mix the deck.
+ */
+exports.mixDeck = function (deck) {
```

Comment:
Can you explain this function ? I don't understand where you constants came from... If they are constants can you add them as
```node
const RANDOM_MIX = 3;
```

Reply:
This is a first draft for a mix function:

Top of the deck is switched with a random card in the deck, n times (where n is the length of the deck).
And theses swaps are repeated k times (where k is the RANDOM_MIX).

Reply:
Ok